Default users 'membership' to 'inclusive'

ckaenzig opened the pull request:
This allows removing users from groups by making the groups array
'inclusive' (user belongs to these groups and no more) instead of
a 'minimum'. It also insures that the users' rights are more tightly managed.

Should this be a configurable default?

mcanevet commented:
@ckaenzig what about a `membership` param defaulting to `undef`?

ckaenzig commented:
@mcanevet That's what we have now in fact. We could set the `membership` parameter for each user individually in hiera. Is that what you mean?

mcanevet commented:
@ckaenzig I'm not sure that hardcoding a default different from the native `user` type in this module is a good idea. We could make it configurable and override the default for our use case.

ckaenzig commented:
@mcanevet I've changed the PR and added a parameter at the module level to use as a default in the user resource. So the default behaviour is now unchanged and can easily be set by site.
